test/boost/cql_query_test.cc

```diff
@@ -4998,3 +4998,22 @@ SEASTAR_TEST_CASE(timeuuid_fcts_prepared_re_evaluation) {
         }
     });
 }
+
+SEASTAR_TEST_CASE(test_parallelized_select_count) {
+    return do_with_cql_env_thread([](cql_test_env& e) {
+        auto& qp = e.local_qp();
+        auto stat_parallelized = qp.get_cql_stats().select_parallelized;
+
+        e.execute_cql("CREATE TABLE tbl (k int, v int, PRIMARY KEY (k));").get();
+        int value_count = 10;
+        for (int i = 0; i < value_count; i++) {
+            e.execute_cql(format("INSERT INTO tbl (k, v) VALUES ({:d}, {:d});", i, i)).get();
+        }
+        auto msg = e.execute_cql("SELECT COUNT(*) FROM tbl;").get();
+        assert_that(msg).is_rows().with_rows({
+            {long_type->decompose(int64_t(value_count))}
+        });
+
+        BOOST_CHECK_EQUAL(stat_parallelized + 1, qp.get_cql_stats().select_parallelized);
+    });
+}
```

Comment:
Future idea: expose statistics as a CQL table, so we can run this test via cql-pytest.

```cql
EXEC reset_statistics()
...
SELECT count(*) FROM tbl;
...
SELECT select_parallized FROM system.cql_statistics;
```

And then expect both the result of the count and the statistics update. It's a little fragile since internal queries may update the statistics.

/cc @kostja @xemul